infinitest-lib/src/main/java/org/infinitest/parser/JavaAssistClass.java

```diff
@@ -59,14 +60,34 @@ public class JavaAssistClass extends AbstractJavaClass {
 
 	public JavaAssistClass(CtClass classReference) {
 		imports = findImports(classReference);
-		isATest = !isAbstract(classReference) && hasSupportedTests(classReference);
+		isATest = !isAbstract(classReference) &&
+				(hasTests(classReference) || isJUnit5Testable(classReference)) &&
+				(hasJUnit5TestImport(imports) || canInstantiate(classReference));
 		className = classReference.getName();
 	}
 
-	private boolean isJUnit5TestClass(final String[] imports) {
+	private boolean hasJUnit5TestImport(final String[] imports) {
 		return stream(imports)
 				.anyMatch(org.junit.jupiter.api.Test.class.getName()::equals);
```

Comment:
It's outside a scope of this PR, but probably `@Testable` could be used also for JUnit Jupiter tests.